### ISSUE TITLE
Add `onKeyUp` and` onKeyDown` support?

### ISSUE THREAD

**Issue by alexandergunnarson** [necolas/react-native-web#791]
React Native only supports `onKeyPress` in `TextInput`s (and by extension probably everywhere else, though I haven't looked thoroughly enough). This is fine for devices that don't have keyboards (e.g. phones), but not necessarily good for devices that do. For instance:

1) I can't `.preventDefault` on arrow keys (or other non-printable keys that cause default behavior to happen, though those are less important to me) with `onKeyPress`; for it to actually override default behavior, that needs to happen in an `onKeyDown` handler. See https://github.com/necolas/react-native-web/pull/540 for related use cases.
  - a) For instance, I'm trying to create a spreadsheet app in which the user should be able to navigate easily with arrow keys, but currently the up arrow key causes the selection to go to the beginning of the input and the down arrow key causes it to go to the end; only after this does the focus move to the cell above or below. Only a `.preventDefault` in `onKeyDown` can remedy this.
2) Sometimes a user should be able to hold down a key without releasing it, and still have it trigger an event. Use cases include: 
  - b) A modal or other UI element which appears only as long as a user holds down a particular key
  - c) When a user should be able to hold a printable key (e.g. a space) down in an input and have it continue to append

`onKeyPress` is fine for most use cases, but as you can see, it is insufficient for some. I propose two possible solutions:

- Solution 1) That `TextInput`s (and perhaps other components, but for now at least that one) allow web-specific custom `onKeyDown` and/or `onKeyUp` handlers and ensure that `onKeyPress` is as close to the React Native behavior as possible (as it currently is, i think). I know that `onKeyPress` interacts with both of these handlers to an extent and I'm not sure what the right behavior is, but it seems that the `onKeyDown` and `onKeyUp` handlers, if present, should always fire when the DOM fires their respective events.
- Solution 2) (Insufficient for case 2a) but sufficient for the others) Edit the following line to add support for other non-printable keys (such as at least the arrow keys) that, like Backspace and Tab, do not fire `keyPress` events:
https://github.com/necolas/react-native-web/blob/31d428a649ecf268628d6f6622a99f51e7344b1b/packages/react-native-web/src/exports/TextInput/index.js#L304

**Comment by necolas** [comment]
I'm finding this a bit hard to follow.

> I can't `preventDefault` on "Enter" or "Tab" with `onKeyPress`; for it to actually override default behavior, that needs to happen in an `onKeyDown` handler

In those cases, `onKeyPress` is equivalent to the `onKeyDown` handler: https://github.com/necolas/react-native-web/blob/master/packages/react-native-web/src/exports/TextInput/index.js#L299-L307

> A modal or other UI element which appears only as long as a user holds down a particular key

How is this related to `TextInput`?

> Edit the following line to add support for other non-printable keys (such as at least the arrow keys)

That seems fine for now unless I can understand the need for anything more.

**Comment by alexandergunnarson** [comment]
You're right — the comment about "Enter" or "Tab" is outdated; I wrote it before I had explored the code sufficiently. I should have said instead:

> I can't preventDefault on the arrow keys (or other non-printable keys that cause default behavior to happen, though those are less important to me) with `onKeyPress`; for it to actually override default behavior, that needs to happen in an `onKeyDown` handler.

> A modal or other UI element which appears only as long as a user holds down a particular key

This isn't especially important for my use case so we can consider it out of scope for my purposes. That said, it is within the scope of this issue. Imagine a scenario where you want a custom behavior similar to Mac OS and iOS's behavior where you hold down e.g. the "A" key in an input and above it shows possibilities for accented versions of that character which you can navigate to using arrow keys. Perhaps you'd like to style such UI elements to have a consistent look and feel with your app rather than relying on the OS to implement the feature for you (which, for instance, Windows doesn't, and probably neither do older versions of Mac OS) and show it at the appropriate time.

**Comment by alexandergunnarson** [comment]
Heh, thanks for taking care of that! I've been short on time lately.
